pkg/services/validator/validator.go

```diff
@@ -78,28 +80,88 @@ func (s *service) Configure() (err error) {
 	}
 
 	var (
-		irmaConfig *irma.Configuration
+		irmaConfig *irmago.Configuration
 		irmaServer *irmaserver.Server
 	)
 
 	if irmaServer, irmaConfig, err = s.configureIrma(s.config); err != nil {
 		return
 	}
 
-	irmaService := irmaService.IrmaService{
-		IrmaSessionHandler: &irmaService.DefaultIrmaSessionHandler{I: irmaServer},
+	irmaService := irma.Service{
+		IrmaSessionHandler: &irma.DefaultIrmaSessionHandler{I: irmaServer},
 		IrmaConfig:         irmaConfig,
-		Registry:           s.registry,
-		Crypto:             s.crypto,
-		ContractTemplates:  s.contractTemplates,
+		// todo: remove this when the deprecated irmaValidatorValidateJwt is removed
+		Registry:          s.registry,
+		Crypto:            s.crypto,
+		IrmaServiceConfig: s.irmaServiceConfig,
 	}
+	// todo refactor and use signer/verifier
 	s.contractSessionHandler = irmaService
 	s.contractValidator = irmaService
+
+	s.verifiers = map[string]contract.Verifier{}
+	s.signers = map[string]contract.Signer{}
+
+	cvMap := make(map[string]bool, len(s.config.ContractValidators))
+	for _, cv := range s.config.ContractValidators {
+		cvMap[cv] = true
+	}
+
+	// todo config to VP types
+	if _, ok := cvMap["irma"]; ok {
+		s.verifiers[irma.VerifiablePresentationType] = irmaService
+		s.signers[irma.ContractFormat] = irmaService
+	}
+
+	if _, ok := cvMap["dummy"]; ok && !core.NutsConfig().InStrictMode() {
+		d := dummy.Dummy{
+			Sessions: map[string]string{},
+			Status:   map[string]string{},
+		}
+		s.verifiers[dummy.VerifiablePresentationType] = d
+		s.signers[dummy.ContractFormat] = d
+	}
+
 	return
 }
 
-func (s *service) ContractValidatorInstance() services.ContractValidator {
-	return s.contractValidator
+func (s *service) VerifyVP(rawVerifiablePresentation []byte) (*contract.VerificationResult, error) {
+	vp := contract.BaseVerifiablePresentation{}
+	if err := json.Unmarshal(rawVerifiablePresentation, &vp); err != nil {
+		return nil, fmt.Errorf("unable to verifyVP: %w", err)
+	}
+
+	// remove default type
+	types := vp.Type
+	n := 0
+	for _, x := range types {
+		if x != contract.VerifiablePresentationType {
+			types[n] = x
+			n++
+		}
+	}
+	types = types[:n]
+
+	if len(types) != 1 {
+		return nil, errors.New("unprocessable VerifiablePresentation, exactly 1 custom type is expected")
+	}
+	t := types[0]
+
+	if _, ok := s.verifiers[t]; !ok {
+		return nil, fmt.Errorf("unknown VerifiablePresentation type: %s", t)
+	}
+
+	return s.verifiers[t].VerifyVP(rawVerifiablePresentation)
+}
+
+func (s *service) SigningSessionStatus(sessionID string) (contract.SigningSessionResult, error) {
+	for _, signer := range s.signers {
```

Comment:
should you give it to the signer and ask "can you process this?" and process it if it can?

What if 2 implementations use an integer sessionID internally so the first one says "hey I have found it!" but it really belonged to the second impl.

Reply:
sessionIDs should be unique (security reasons). The problem is that an identifier on its own doesn't say anything about which signer to use. So trying is one option (collisions are not 2 b expected), returning the means as well is another, but makes the API less elegant.